src/main/java/org/lightcouch/CouchDbClient.java

```diff
@@ -205,6 +206,8 @@ private PoolingHttpClientConnectionManager createConnectionManager(
 
 		if("https".equals(props.getProtocol())) {
 			if (props.isSSLAuthenticationDisabled()) {
+				// No SSL authentication.
+				// No need for a custom SSLSocketFactory in this case.
```

Comment:
<i>&gt; this needs at least documenting in the public documentation</i>
Yes, thanks for noticing this. I've added a commit "Javadoc tweak" to this branch.

<i>&gt; if not including checks for in the setters in ConnectOptions.</i>
Such checks would not be in line with Java Bean coding conventions. Java Bean specification implies that the various properties of a Java Bean are *independent* - even if afterwards their effects/semantics are interdependent  -, which implies that an application can set the different properties in arbitrary order. This guarantee that the order in which you call the set... methods is arbitrary is not only important for GUI applications that shows a property dialog, but also for the normal programmer who does not expect that the order matters. So,
<pre>options.setSSLAuthenticationDisabled(true);
options.setSecureSSLSocketFactory(myFactory);
options.setSSLAuthenticationDisabled(false);
</pre>
should have the same effect as
<pre>options.setSSLAuthenticationDisabled(true);
options.setSSLAuthenticationDisabled(false);
options.setSecureSSLSocketFactory(myFactory);
</pre>
and the last of these code snippets definitely should not produce an exception.